File: add-gif-urls.py
import re

# Read the file
with open('src/data/exerciseLibraryExpanded.js', 'r', encoding='utf-8') as f:
    content = f.read()

# Find all exercise blocks
# Pattern: icon: '...' followed by newline and closing brace
pattern = r"(\s+id: '([^']+)',[\s\S]*?icon: '[^']+')(\n\s+})"

def replacement(match):
    before_icon = match.group(1)
    exercise_id = match.group(2)
    after_icon = match.group(3)
    
    # Check if gifUrl already exists in this exercise
    if 'gifUrl' in before_icon:
        return match.group(0)
    
    # Add gifUrl before the closing brace
    return f"{before_icon},\n    gifUrl: '/assets/exercise-gifs/{exercise_id}.gif'{after_icon}"

# Apply the replacement
updated_content = re.sub(pattern, replacement, content)

# Write back
with open('src/data/exerciseLibraryExpanded.js', 'w', encoding='utf-8') as f:
    f.write(updated_content)

# Count results
gif_count = updated_content.count('gifUrl:')
print(f"✅ Added gifUrl to exercises!")
print(f"📊 Total exercises with gifUrl: {gif_count}")
print(f"")
print(f"Next steps:")
print(f"1. Download exercise GIFs from Pexels/Pixabay")
print(f"2. Place in: public/assets/exercise-gifs/")
print(f"3. Name them: chest-1.gif, chest-2.gif, back-1.gif, etc.")
print(f"4. Build and test!")

File: add-gif-urls.js
// Script to add gifUrl to all exercises in exerciseLibraryExpanded.js
// Run with: node add-gif-urls.js

const fs = require('fs');
const path = require('path');

const filePath = path.join(__dirname, 'src', 'data', 'exerciseLibraryExpanded.js');

console.log('📝 Adding gifUrl to all exercises...');
console.log('File:', filePath);

// Read the file
let content = fs.readFileSync(filePath, 'utf8');

// Pattern to match exercise objects
// Matches: icon: '💪'\n  },
// And adds: icon: '💪',\n    gifUrl: '/assets/exercise-gifs/EXERCISE_ID.gif'\n  },
const pattern = /(\s+id: '([^']+)',[\s\S]*?icon: '[^']+')(\n\s+})/g;

let matchCount = 0;
const updatedContent = content.replace(pattern, (match, beforeIcon, exerciseId, afterIcon) => {
  matchCount++;
  // Check if gifUrl already exists
  if (match.includes('gifUrl:')) {
    return match; // Skip if already has gifUrl
  }
  return `${beforeIcon},\n    gifUrl: '/assets/exercise-gifs/${exerciseId}.gif'${afterIcon}`;
});

// Write back to file
fs.writeFileSync(filePath, updatedContent, 'utf8');

console.log(`✅ Added gifUrl to ${matchCount} exercises!`);
console.log('');
console.log('📦 Next steps:');
console.log('1. Download exercise GIFs from Pexels/Pixabay');
console.log('2. Rename them to match exercise IDs (chest-1.gif, back-1.gif, etc.)');
console.log('3. Place in: public/assets/exercise-gifs/');
console.log('4. Rebuild app: npm run build');
console.log('');
console.log('💡 Fallback: If GIF not found, CSS animation will show automatically!');
